api/src/main/java/com/flipkart/poseidon/pojos/ParamPOJO.java

```diff
@@ -96,6 +104,18 @@ public String getSeparator() {
         return separator;
     }
 
+    public JavaType getJavaType() {
+        return javaType;
+    }
+
+    public void setJavaType(JavaType javaType) {
+        this.javaType = javaType;
+    }
+
+    public VariableModel getType() {
+        return type;
+    }
+
```

Comment:
So, datatype, javatype (deprecated), type are the possible ways to define types?

And a "type" example for defining `Map<String, Integer>` would be
```
"type": {
   "type": "java.util.Map",
   "types": [
      {
          "type": "java.lang.String",
      },
      {
          "type": "java.lang.Integer",
      }
   ]
}
```

Is this right?

Reply:
yes thats correct

Reply:
ok, looks bit confusing even for a simple use case of java types without generics
```
"type": {
   "type": "com.flipkart.poseidon.MyType"
}
```

See if this can be simplified. Worst case, update documentation on how to go about defining types in API json files [here](https://github.com/flipkart-incubator/Poseidon/wiki/api%20(wip)).

Reply:
there's a simpler way of describing this use case: `"type": "com.flipkart.poseidon.MyType"`
This by itself is also enough as Jackson will use the String contructor in the VariableModel class to construct the object

Reply:
and the Map example can be defined as:
`"type": {
   "type": "java.util.Map",
   "types": [
          "java.lang.String",
          "java.lang.Integer"
   ]
}`